app/src/main/java/com/vero/tasky/agenda/presentation/agenda/components/AgendaComponent.kt

```diff
@@ -0,0 +1,92 @@
+package com.vero.tasky.agenda.presentation.agenda.components
+
+import androidx.compose.foundation.background
+import androidx.compose.foundation.clickable
+import androidx.compose.foundation.layout.*
+import androidx.compose.foundation.shape.RoundedCornerShape
+import androidx.compose.material.Icon
+import androidx.compose.material.MaterialTheme
+import androidx.compose.material.Text
+import androidx.compose.material.icons.Icons
+import androidx.compose.material.icons.filled.MoreHoriz
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.clip
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.text.style.TextAlign
+import androidx.compose.ui.unit.dp
+import com.vero.tasky.agenda.domain.model.AgendaItem
+import com.vero.tasky.agenda.presentation.util.AgendaTimeParser
+import com.vero.tasky.ui.theme.*
+
+@Composable
+fun AgendaComponent(
+    agendaItem: AgendaItem,
+    modifier: Modifier = Modifier,
+    onOpenClick: (agendaItem: AgendaItem) -> Unit,
+    onEditClick: (agendaItem: AgendaItem) -> Unit,
+    onDeleteClick: (agendaItem: AgendaItem) -> Unit,
+    onCheckChanged: (agendaItem: AgendaItem.Task) -> Unit,
+    isCurrent: Boolean
+) {
+
+    val backgroundColor = when(agendaItem) {
+        is AgendaItem.Reminder -> MaterialTheme.colors.reminder
+        is AgendaItem.Task -> MaterialTheme.colors.task
+        is AgendaItem.Event -> MaterialTheme.colors.event
```

Comment:
I'd make these extra colors independent of the theme, since the theme is rather used to define different colors for light and dark mode, but these don't differ